classes/submissionFile/DAO.inc.php

```diff
@@ -0,0 +1,466 @@
+        $excludePubObjectId,
+        $contextId
+    ): bool {
+        $result = DB::table($this->settingsTable . ' as sfs')
+            ->join('submission_files AS sf', 'sfs.submission_file_id', '=', 'sf.submission_file_id')
+            ->join('submissions AS s', 'sf.submission_id', '=', 's.submission_id')
+            ->where([
+                'sfs.setting_name' => 'pub-id::' . (string) $pubIdType,
+                'sfs.setting_value' => (string) $pubId,
+                'sfs.submission_file_id' => (int) $excludePubObjectId,
+                's.context_id' => (int) $contextId
+            ])->count();
+        return (bool) $result > 0;
+    }
+
+    /**
+     * @copydoc PKPPubIdPluginDAO::changePubId()
+     */
+    public function changePubId($pubObjectId, $pubIdType, $pubId)
+    {
+        DB::table($this->settingsTable)
+            ->where([
+                'submission_file_id' => (int) $pubObjectId,
+                'setting_name' => 'pub-id::' . (string) $pubIdType,
+                'setting_value' => (string) $pubId
+            ])
+            ->update([
+                'submission_file_id' => (int) $pubObjectId,
+                'locale' => '',
+                'setting_name' => 'pub-id::' . $pubIdType,
+                'setting_type' => 'string',
+                'setting_value' => (string) $pubId
+            ]);
+    }
+
+    /**
+     * @copydoc PKPPubIdPluginDAO::deletePubId()
+     */
+    public function deletePubId($pubObjectId, $pubIdType)
+    {
+        DB::table($this->settingsTable)
+            ->where([
+                'submission_file_id' => (int) $pubObjectId,
+                'setting_name' => 'pub-id::' . (string) $pubIdType
+            ])->delete();
+    }
+
+    /**
+     * @copydoc PKPPubIdPluginDAO::deleteAllPubIds()
+     */
+    public function deleteAllPubIds($contextId, $pubIdType)
+    {
+        $collector = Repo::submission()
+            ->getCollector()
+            ->filterByContextIds([$contextId]);
+        $submissions = Repo::submission()->getMany($collector);
+        $submissionsIds = [];
+        foreach ($submissions as $submission) {
+            $submissionsIds[] = $submission->getId();
+        }
+
+        $submissionFilesCollector = Repo::submissionFiles()
+            ->getCollector()
+            ->filterBySubmissionIds($submissionsIds);
+
+        Repo::submissionFiles()
+            ->deleteMany($submissionFilesCollector);
+    }
+
+    /**
+     * Insert a review round for Submission File
+     *
+     * @throws Exception If we couldn't find the review Round, throws an exception.
+     */
+    protected function insertReviewRound(SubmissionFile $submissionFile): void
+    {
+        if (
+            in_array(
+                $submissionFile->getData('assocType'),
+                [
+                    Application::ASSOC_TYPE_REVIEW_ROUND,
+                    Application::ASSOC_TYPE_REVIEW_ASSIGNMENT
+                ]
+            )
+        ) {
+            return;
```

Comment:
Did the original logic for this accidentally get removed? See `PKPSubmissionFileDAO::insertObject`.

Reply:
Small typo, forgot to add the `!` into the `in_array` in the previous commit.